### PR TITLE
NIFI-4506 Adding toDate and format functions to record path

### PR DESCRIPTION
**Pull request by bbende**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/paths/RecordPathCompiler.java [hunk 1]
```diff
@@ -346,4 +356,19 @@ private static RecordPathFilter createFunctionFilter(final Tree functionTree, fi
 
         return argPaths;
     }
+
+    private static RecordPathSegment[] getArgPaths(final Tree argumentListTree, final int minCount, final int maxCount, final String functionName, final boolean absolute) {
```
Comment (mattyb149) [inline nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/paths/RecordPathCompiler.java]: Is this used anywhere?

**Comment by pvillard31** [comment]
Just a quick comment: would it make sense to also add support for time zone as we do in expression language? Can be done in a separate JIRA though.

**Comment by bbende** [comment]
Yea I forgot about the timezone, but that should be added as an optional parameter, and that will make use of the unused method that Matt pointed out :)

**Comment by bbende** [comment]
I'd like to merge this in as is (barring any feedback) and add the timezone support later. 

I created this JIRA for the timezone support: https://issues.apache.org/jira/browse/NIFI-4732

**Comment by markap14** [comment]
@bbende sounds reasonable to me. Will review as-is for now.

**Comment by mattyb149** [comment]
@bbende works for me too. Mark will review as-is for now ;)

**Comment by markap14** [comment]
All looks good to me! +1 merged to master.

**Pull request closed without merging**